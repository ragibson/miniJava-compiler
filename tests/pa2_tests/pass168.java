// PA1 parse decl pass
class Test {

    void p() {
	int a = 3;
	int [] b = new int[4];
    }
}

