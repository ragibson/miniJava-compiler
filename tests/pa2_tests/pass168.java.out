********** miniJava Compiler (PA2) **********
Syntactic Analysis ... 
Syntactic Analysis OK
======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "Test" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   VOID BaseType
  .   .   "p" methodname
  .   .   ParameterDeclList [0]
  .   .   StmtList [2]
  .   .   . VarDeclStmt
  .   .   .   VarDecl
  .   .   .     INT BaseType
  .   .   .     "a" varname
  .   .   .   LiteralExpr
  .   .   .     "3" IntLiteral
  .   .   . VarDeclStmt
  .   .   .   VarDecl
  .   .   .     ArrayType
  .   .   .       INT BaseType
  .   .   .     "b" varname
  .   .   .   NewArrayExpr
  .   .   .     INT BaseType
  .   .   .     LiteralExpr
  .   .   .       "4" IntLiteral
=============================================
